simplify logfile rotation

we simply rename <logfile> to <logfile>.1. That way we do not need
any logrotate script. max_logfile is set to 256MB.


Index: new/lib/logger.c
===================================================================
--- new.orig/lib/logger.c	2013-02-28 07:15:26.000000000 +0100
+++ new/lib/logger.c	2013-02-28 07:29:55.000000000 +0100
@@ -75,7 +75,7 @@
 static key_t semkey;
 static char *log_buff;
 
-static int64_t max_logsize = 500 * 1024 * 1024;  /*500MB*/
+static int64_t max_logsize = 256 * 1024 * 1024;  /*500MB*/
 
 static pthread_mutex_t logsize_lock = PTHREAD_MUTEX_INITIALIZER;
 
@@ -243,13 +243,7 @@
 
 	if (access(log_nowname, R_OK) == 0) {
 		char old_logfile[256];
-		time_t t;
-		struct tm tm;
-		time(&t);
-		localtime_r((const time_t *)&t, &tm);
-		sprintf(old_logfile, "%s.%04d-%02d-%02d-%02d-%02d",
-				log_nowname, tm.tm_year + 1900, tm.tm_mon + 1,
-				tm.tm_mday, tm.tm_hour, tm.tm_min);
+		sprintf(old_logfile, "%s.1", log_nowname);
 		rename(log_nowname, old_logfile);
 	}
 	new_fd = open(log_nowname, O_RDWR | O_CREAT | O_APPEND, 0644);
